From e7c8c3e795cd5c663b85d467b90e6dd36068da81 Mon Sep 17 00:00:00 2001
From: Nishanth Menon <nm@ti.com>
Date: Fri, 3 May 2024 18:15:31 -0500
Subject: [PATCH 2/5] HACK: drm/bridge: ite-it66121: Hack hpd back in

Commit 85e6b8d401de ("HACK: drm/bridge: ite-it66121: Add drm_connector
support") and related patches seem to have broken HPD. hack the same
in.

NOTE: This hack is not perfect, if the HDMI is not plugged in at the
very start of the boot, we seem to manually need to trigger the
following (path to display depends on platform and configuration)

echo connected|sudo tee /sys/class/drm/card0-HDMI-A-1/status

Signed-off-by: Jared McArthur <j-mcarthur@ti.com>
Signed-off-by: Nishanth Menon <nm@ti.com>
---
 drivers/gpu/drm/bridge/ite-it66121.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/gpu/drm/bridge/ite-it66121.c b/drivers/gpu/drm/bridge/ite-it66121.c
index 695f93b280da..d6cf6db305d6 100644
--- a/drivers/gpu/drm/bridge/ite-it66121.c
+++ b/drivers/gpu/drm/bridge/ite-it66121.c
@@ -674,6 +674,8 @@ static const struct drm_connector_funcs it66121_connector_funcs = {
 	.atomic_destroy_state = drm_atomic_helper_connector_destroy_state,
 };
 
+static void it66121_bridge_hpd_enable(struct drm_bridge *bridge);
+
 static int it66121_bridge_attach(struct drm_bridge *bridge,
 				 enum drm_bridge_attach_flags flags)
 {
@@ -758,6 +760,9 @@ static int it66121_bridge_attach(struct drm_bridge *bridge,
 	/* Per programming manual, sleep here for bridge to settle */
 	msleep(50);
 
+	/* HACK For some part of missing hotplug somewhere.. */
+	it66121_bridge_hpd_enable(bridge);
+
 	return 0;
 }
 
-- 
2.47.2

